cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -540,12 +541,11 @@ func (c *machineController) nodeGroupForNode(node *corev1.Node) (*nodegroup, err
 		return nil, fmt.Errorf("failed to build nodegroup for node %q: %v", node.Name, err)
 	}
 
-	// We don't scale from 0 so nodes must belong to a nodegroup
-	// that has a scale size of at least 1.
-	if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
-		return nil, nil
+	if !nodegroup.scalableResource.CanScaleFromZero() {
+		if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
```

Comment:
shouldn't this be always the cases regardless of `CanScaleFromZero`? max-min must be at least 1.

Reply:
https://github.com/openshift/kubernetes-autoscaler/pull/137/files#r394255920

Reply:
your reasoning makes sense to me here, i'm going to drop the check against CanScaleFromZero and just rely on the MaxSize-MinSize > 1.

Reply:
one question though, should we check for the situation where `MaxSize-MinSize < 1 && MaxSize != 0` ?

essentially, do we need to return the nodegroup when max and min are equal, but not zero?

_i'm guessing not, since we won't be autoscaling in those cases, but i want to make sure_